endpoints-framework/src/main/java/com/google/api/server/spi/config/jsonwriter/JsonConfigWriter.java

```diff
@@ -171,6 +171,12 @@ private void convertApi(ObjectNode root, ApiConfig config) {
     if (config.getDocumentationLink() != null) {
       root.put("documentation", config.getDocumentationLink());
     }
+    if (config.getIconX16() != null || config.getIconX32() != null) {
+      ObjectNode icons = objectMapper.createObjectNode();
+      icons.put("x16", config.getIconX16());
+      icons.put("x32", config.getIconX32());
```

Comment:
Shouldn't we handle the case were just one of them has been provided ?
```
if (config.getIconX16() != null) {
  icons.put("x16", config.getIconX16());
}
if (config.getIconX32() != null) {
  icons.put("x32", config.getIconX32());
}
```

Reply:
I updated the code to require both icon sizes to be set (don't want inconsistent display).
This is not a blocking error however to set only one version though.